components/automate-ui/src/app/pages/+compliance/+reporting/reporting.component.spec.ts

```diff
@@ -84,9 +84,9 @@ describe('ReportingComponent', () => {
     ['00', 'exactly end of day'],
     ['01', 'an hour after end of day']
   ], function (hour: string, description: string) {
-    it(`maps hour correctly without timezone for ${description}`, () => {
+    it(`maps hour correctly to UTC timestandard for ${description}`, () => {
```

Comment:
Right now this is a phantom test: we have not demonstrated that the fix does anything useful, simply because the test passed before the fix and the test _without change_ still passes after the fix. What we need to do is revise the set of tests so that:
(a) some of the tests fail with the original code under test, and
(b) all of the tests pass with the fix in place.
The set of equivalence classes already present are looking at the boundaries of the end-of-the-day but are oblivious to UTC boundaries. (Indeed, if you examine `dateBefore` it shows `isUTC` is false, while in the production code `convertMomentToDate` will be called with `isUTC` set true.)
Ultimately: Should the hour be coming back the same as the input, or should it be adjusted by the UTC offset?

Furthermore, this set of tests is only examining the hour.  what we are more concerned with in this PR is the month and day. We should have tests, for example, that emulate the Australian situation, coming in with e.g. July 2 and getting July 1 returned (and a nice boundary value of coming in with July 1 and getting June 30 returned).

And a minor point: `dateBefore` and `dateAfter` lead one to think (incorrectly) that the dates are sequenced. I would suggest `originalDate` and `resultDate` or some such.

Reply:
@msorens - This is still a phantom test but i'm trying to wrap my head around the best way to do this.

The reason we were seeing a day behind is interesting but here's a small example: https://stackoverflow.com/questions/17855842/moment-js-utc-gives-wrong-date

Simply, When the new date was being created, it was done in local time which sets the hours to 00 and then gives us a day behind when formatted.

I'd appreciate if you might help me figure out the best way to test this, because the issue doesn't present itself until its actually formatted, so the tests will pass for the current output, and for the fix output.  I thought I would just use the DatetimePipe in the test, but it didn't like that for some reason.